### DISCUSSION TITLE
New build not picking up branches

### DISCUSSION THREAD

**Discussion by d7p** [codemagic-ci-cd/codemagic-docs#1919] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Go to application list page
2. click 'start new build' on the application

The 'popup'/ dialog box shows a spinner to branches are loading branches but never loads any including main. There is no error or time out just the dialogue box sitting there. 

### Expected results

Dialog box loads current branches and lists them under the 'Select branch' text and then can start a build.

### Actual results

No branches are load and no error is shown

**Comment by himesh-cm** [comment, marked as answer]
Hi @d7p,I have tested it, and it seems to be working fine. 
Could you kindly check the git integration in your `teams>settings` and try reconnecting to see if it resolves the issue? If not, please share your team ID from the `teams>settings` URL.

**Comment by d7p** [comment]
thank you this worked. 
I also tried to update the repository URL and get an error there before reconnecting it to gitlab. It would be good to get a error on with the start new build dialog if it can't connect
